### PR TITLE
Trust search UI only

### PR DESCRIPTION
**Pull request by MarkRobinsonDfe**
Implement trust search UI:-
1) New ApplicationSelectTrust page
2) New _TrustDetails partial view to display selected trust
3) academies.trustsearch.js - to do heavy lifting between UI & backend
4) TrustController - to wire up to trams API
5) New ApplicationSelectTrust page unit tests

To review HTML use following test URL:-
https://localhost:44350/trust/application-select-trust?appId=0

### CODE REVIEW

File: Dfe.Academies.External.Web/AcademiesAPIResponseModels/Trusts/TrustSearchDto.cs [hunk 1]
```diff
@@ -2,6 +2,19 @@
 
 public record TrustSearchDto
 {
+	public TrustSearchDto(string ukprn,
```
Comment (wystanpalm-madetech) [inline Dfe.Academies.External.Web/AcademiesAPIResponseModels/Trusts/TrustSearchDto.cs:5]: We've been using the constructor-only syntax for records, which looks like:

```
public record Person(string FirstName, string LastName);
```

Just a thought
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record

File: Dfe.Academies.External.Web.UnitTest/Controllers/SchoolControllerTests.cs [hunk 2]
```diff
@@ -45,8 +44,7 @@ public async Task SchoolController___ReturnSchoolDetailsPartialViewPopulated___R
 	    // arrange
 	    var mockLogger = new Mock<ILogger<SchoolController>>();
 	    var mockReferenceDataRetrievalService = new Mock<IReferenceDataRetrievalService>();
-	    var mockConversionApplicationRetrievalService = new Mock<IConversionApplicationRetrievalService>();
-        var schoolController = new SchoolController(mockLogger.Object, mockReferenceDataRetrievalService.Object, mockConversionApplicationRetrievalService.Object);
+        var schoolController = new SchoolController(mockLogger.Object, mockReferenceDataRetrievalService.Object);
```
Comment (wystanpalm-madetech) [inline Dfe.Academies.External.Web.UnitTest/Controllers/SchoolControllerTests.cs:47]: Can we have the test name:

```
ReturnSchoolDetailsPartialViewPopulated___[Scenario]___ReturnsPartialView
```

I'm not quite sure what the scenario is here, "SchoolSelected" maybe, as opposed to "SchoolNull"?

File: Dfe.Academies.External.Web.UnitTest/Controllers/TrustControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Dfe.Academies.External.Web.UnitTest.Controllers;
+
+[Parallelizable(ParallelScope.All)]
+internal sealed class TrustControllerTests
+{
+    [Test]
+    public async Task TrustController___Search___ReturnsResult()
```
Comment (wystanpalm-madetech) [inline Dfe.Academies.External.Web.UnitTest/Controllers/TrustControllerTests.cs:22]:
```
Seach___ResultsFound___ResultsReturned()
```

File: Dfe.Academies.External.Web.UnitTest/Controllers/TrustControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        // act
+        var result = await trustController.Search(trustName);
+
+        // assert
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result.Count(), Is.Zero);
+    }
+
+    [Test]
+    public async Task TrustController___ReturnSchoolDetailsPartialViewPopulated___ReturnsPartialView()
```
Comment (wystanpalm-madetech) [inline Dfe.Academies.External.Web.UnitTest/Controllers/TrustControllerTests.cs:45]:
```
ReturnSchoolDetailsPartialViewPopulated___[Scenario]___ReturnsPartialView
```

Is the scenario "TrustNamePassed"?

**Review by wystanpalm-madetech — CHANGES_REQUESTED** [review]
A few small change requests, mostly just test names.

File: Dfe.Academies.External.Web.UnitTest/Controllers/SchoolControllerTests.cs (hunk 2 above)
Comment (MarkRobinsonDfe) [reply Dfe.Academies.External.Web.UnitTest/Controllers/SchoolControllerTests.cs:47]: Amended name in PR68 as that's where the school controller tests are contained

File: Dfe.Academies.External.Web/AcademiesAPIResponseModels/Trusts/TrustSearchDto.cs (hunk 1 above)
Comment (MarkRobinsonDfe) [reply Dfe.Academies.External.Web/AcademiesAPIResponseModels/Trusts/TrustSearchDto.cs:5]: I need the DisplayName prop though for the autocomplete

**Review by wystanpalm-madetech — APPROVED** [review]

**Review by wystanpalm-madetech — APPROVED** [review]
There seem to be two branches here that have been merged into each other, it's probably too much hassle to try and unpick what's happened there so I'm approving.

**Pull request merged**